src/sql/workbench/parts/query/common/queryInput.ts

```diff
@@ -301,9 +301,13 @@ export class QueryInput extends EditorInput implements IEncodingSupport, IConnec
 		this.state.connected = false;
 	}
 
-	public onConnectCanceled(): void {
+	public onConnectCanceled(skipConnectedStateReset?: boolean): void {
+		// If we're currently connecting and then cancel, set connected state to false
```

Comment:
It would be great to add a couple unit tests under [https://github.com/microsoft/azuredatastudio/tree/823d136a00cbeb5764f1fcbea527351a2067f532/src/sql/workbench/parts/query/test](https://github.com/microsoft/azuredatastudio/tree/823d136a00cbeb5764f1fcbea527351a2067f532/src/sql/workbench/parts/query/test)

Reply:
Tracked in #7281